FTBFS boinc-client 7.22.0: gzread

**Describe the bug** (Observed behavior)

As part of the make routine of the just-released 7.22.0:

```
[   35s] libtool: link: DIE_RPATH_DIE="/usr/lib64:" /usr/bin/g++ -Wall -Wextra -Wshadow -Wredundant-decls -Wdisabled-optimization -Wpointer-arith -
Wstrict-aliasing -Wcast-align -I/usr/include -I/usr/include/openssl -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwi
nd-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -ffat-lto-objects -g -W -pipe -fno-strict-aliasing -D_REENTRANT
 -Wall -ffast-math -O3 -ffat-lto-objects -o .libs/boinc_client boinc_client-acct_mgr.o boinc_client-acct_setup.o boinc_client-app.o boinc_client-ap
p_config.o boinc_client-app_control.o boinc_client-app_start.o boinc_client-async_file.o boinc_client-check_state.o boinc_client-client_msgs.o boin
c_client-client_state.o boinc_client-client_types.o boinc_client-coproc_sched.o boinc_client-cpu_sched.o boinc_client-cs_account.o boinc_client-cs_
apps.o boinc_client-cs_benchmark.o boinc_client-cs_cmdline.o boinc_client-cs_files.o boinc_client-cs_notice.o boinc_client-cs_platforms.o boinc_client-cs_prefs.o boinc_client-cs_proxy.o boinc_client-cs_scheduler.o boinc_client-cs_statefile.o boinc_client-cs_trickle.o boinc_client-current_version.o boinc_client-dhrystone.o boinc_client-dhrystone2.o boinc_client-file_names.o boinc_client-file_xfer.o boinc_client-gpu_amd.o boinc_client-gpu_detect.o boinc_client-gpu_intel.o boinc_client-gpu_nvidia.o boinc_client-gpu_opencl.o boinc_client-gui_http.o boinc_client-gui_rpc_server.o boinc_client-gui_rpc_server_ops.o boinc_client-hostinfo_linux.o boinc_client-hostinfo_network.o boinc_client-http_curl.o boinc_client-log_flags.o boinc_client-mac_address.o boinc_client-main.o boinc_client-net_stats.o boinc_client-pers_file_xfer.o boinc_client-project.o boinc_client-project_list.o boinc_client-result.o boinc_client-rr_sim.o boinc_client-sandbox.o boinc_client-scheduler_op.o boinc_client-thread.o boinc_client-time_stats.o boinc_client-whetstone.o boinc_client-work_fetch.o boinc_client-hostinfo_unix.o  -L../lib ../lib/.libs/libboinc.so -L/usr/lib64 ../lib/.libs/libboinc_crypt.so -lcurl -lssl -lcrypto -ldl -lfreetype -lresolv -lpthread -lm
[   35s] /usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: boinc_client-async_file.o: undefined reference to symbol 'gzread'
[   35s] /usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libz.so.1: error adding symbols: DSO missing from command line
```

**Steps To Reproduce**

1. I _think_ it may involve building with `-Wl,--as-needed`.

**Expected behavior**

Include what you use. Let the build succeed.

```
boinc_client_LDADD = $(LIBBOINC) $(LIBBOINC_CRYPT) $(BOINC_EXTRA_LIBS) $(PTHREAD_LIBS)
```

This line is missing -lz (or a suitable call to pkgconfig zlib).

**System Information**

 - OS: openSUSE Tumbleweed 20230122
 - BOINC Version: 7.22.0

--- a/configure.ac
+++ b/configure.ac
@@ -547,8 +547,14 @@ SAH_CHECK_LIB([freetype], [fopen],
     [BOINC_EXTRA_LIBS="${BOINC_EXTRA_LIBS} ${sah_lib_last}"])
 SAH_CHECK_LIB([socket], [bind],
     [BOINC_EXTRA_LIBS="${BOINC_EXTRA_LIBS} ${sah_lib_last}"])
-SAH_CHECK_LIB([z], [gzopen],
-    [BOINC_EXTRA_LIBS="${BOINC_EXTRA_LIBS} ${sah_lib_last}"])
+SAH_CHECK_LIB([z], [gzopen], [
+      zlib_found=yes
+      BOINC_EXTRA_LIBS="${BOINC_EXTRA_LIBS} ${sah_lib_last}"])
+if test "x${zlib_found}" = "x" -a "${isWIN32}" = "no"; then
+    if test "x${enable_client}" = "xyes" -o "x${enable_server}" = "xyes" -o "x${enable_apps_vbox}" = "xyes"]; then
+        AC_MSG_ERROR([zlib is required but was not found])
+    fi
+fi
 SAH_CHECK_LIB([cups], [md5_finish],
     [BOINC_EXTRA_LIBS="${BOINC_EXTRA_LIBS} ${sah_lib_last}"])
 if test "${isWIN32}" = "yes" ; then

